packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -141,7 +141,18 @@ export default util.createRule<Options, MessageIds>({
       },
       BinaryExpression: binaryExp,
       CallExpression: callExp,
-      // ClassDeclaration
+      ClassDeclaration(node) {
+        if (node.superClass?.type === AST_NODE_TYPES.TSAsExpression) {
+          return rules.ClassDeclaration({
+            ...node,
+            superClass: {
+              ...node.superClass,
+              type: AST_NODE_TYPES.SequenceExpression as any,
+            },
+          });
+        }
+        return rules.ClassDeclaration(node);
```

Comment:
I agree with the Codecov report - we're going to want to test the case where the node doesn't have a super class that is an `as` expression. Both when the super exists as a different type of node, and when there isn't one at all.

Reply:
Oh, I see you're still working on this - sorry, I assumed since it wasn't marked as draft it was ready for review. I'll hold off until explicitly re-requested. 😄

Reply:
@JoshuaKGoldberg Sorry. I should make a draft pr.